rest/diagnostic_api.go

```diff
@@ -0,0 +1,47 @@
+//  Copyright 2013-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+import (
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
+)
+
+type allChannels struct {
+	Channels base.Set `json:"all_channels,omitempty"`
+}
+
+func (h *handler) handleGetAllChannels() error {
+	h.assertAdminOnly()
+	user, err := h.db.Authenticator(h.ctx()).GetUser(internalUserName(mux.Vars(h.rq)["name"]))
+	if err != nil {
+		return err
+	}
+	if user == nil {
+		return kNotFoundError
+	}
+	info := marshalPrincipal(h.db, user, true)
```

Comment:
Since marshalPrincipal is already used by `GET /{db}/_user` and includes channels inherited from roles in the response, why do we need a new endpoint for this functionality?

Reply:
I think the point was to avoid the extra processing done in the GET user handler, I've also wrote a marshalChannels function to avoid the extra processing in marshalPrincipal

Reply:
Also from Jacques

> Keep it in one place
Different permissions may apply that will let support access this endpoint
Potentially add more things to this endpoint down the road to aid in diagnostics without affecting the usual EPs